File: src/Application Framework/Entities/Picklists/PickListValidator.vb
Namespace Entity

    Namespace PickLists

        Public Class PickListValidator : Inherits BaseValidator

            Public Sub Validate(ByVal picklist As Entity.PickLists.PickList)
                If picklist.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In picklist.Errors
                        If picklist.EnumTypeID = CInt(Entity.Sys.Enums.PickListTypes.VATCodes) Then
                            If de.Key = "PercentageRate" Then
                                Me.AddCriticalMessage("Rate Invalid")
                            Else
                                Me.AddCriticalMessage(CStr(de.Value))
                            End If
                        Else
                            Me.AddCriticalMessage(CStr(de.Value))
                        End If
                    Next
                End If

                If picklist.Name.Trim.Length = 0 Then
                    Me.AddCriticalMessage("Name Missing")
                Else
                    If picklist.EnumTypeID = Entity.Sys.Enums.PickListTypes.PartBinReferences Then
                        If DB.Picklists.Check_Unique_Name(picklist.Name, picklist.ManagerID, picklist.EnumTypeID) = False Then
                            Me.AddCriticalMessage("Name already exists")
                        End If
                    End If
                End If
                If picklist.EnumTypeID = 0 Then
                    Me.AddCriticalMessage("Type Missing")
                End If
                If picklist.EnumTypeID = Entity.Sys.Enums.PickListTypes.VATCodes Then
                    If picklist.PercentageRate < 0 Or picklist.PercentageRate > 100 Then
                        Me.AddCriticalMessage("Rate must be between 0 and 100")
                    End If
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If
            End Sub

        End Class

    End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Framework/Entities/Picklists/PickListValidator.cs
using Microsoft.VisualBasic.CompilerServices;
using System.Collections;

namespace FSM.Entity
{
    namespace PickLists
    {
        public class PickListValidator : BaseValidator
        {
            public void Validate(PickList picklist)
            {
                if (picklist.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in picklist.Errors)
                    {
                        if (picklist.EnumTypeID == Conversions.ToInteger(Sys.Enums.PickListTypes.VATCodes))
                        {
                            if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(de.Key, "PercentageRate", false)))
                            {
                                AddCriticalMessage("Rate Invalid");
                            }
                            else
                            {
                                AddCriticalMessage(Conversions.ToString(de.Value));
                            }
                        }
                        else
                        {
                            AddCriticalMessage(Conversions.ToString(de.Value));
                        }
                    }
                }

                if (picklist.Name.Trim().Length == 0)
                {
                    AddCriticalMessage("Name Missing");
                }
                else if (picklist.EnumTypeID == (int)Sys.Enums.PickListTypes.PartBinReferences)
                {
                    if (App.DB.Picklists.Check_Unique_Name(picklist.Name, picklist.ManagerID, (Sys.Enums.PickListTypes)picklist.EnumTypeID) == false)
                    {
                        AddCriticalMessage("Name already exists");
                    }
                }

                if (picklist.EnumTypeID == 0)
                {
                    AddCriticalMessage("Type Missing");
                }

                if (picklist.EnumTypeID == (int)Sys.Enums.PickListTypes.VATCodes)
                {
                    if (picklist.PercentageRate < 0 | picklist.PercentageRate > 100)
                    {
                        AddCriticalMessage("Rate must be between 0 and 100");
                    }
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}